src/braket/pulse/pulse_sequence.py

```diff
@@ -68,6 +68,21 @@ def parameters(self) -> Set[FreeParameter]:
         """Returns the set of `FreeParameter`s in the PulseSequence."""
         return self._free_parameters.copy()
 
+    @property
+    def waveforms(self):
+        """Returns the map of `Waveform` IDs to `Waveform`s in the PulseSequence."""
+        return self._waveforms
+
+    @property
+    def frames(self):
+        """Returns the map of `Frame` IDs to `Frame`s in the PulseSequence."""
+        return self._frames
```

Comment:
These are properties that aren't currently updated if the underlying program is changed. Would that cause confusion?

Reply:
I want to avoid exposing `@<prop>.setter` for these and only allow updates through functions such as `set_frequency()` so that program changes are also done in line with changes to these props.